tweets.py

```diff
@@ -54,52 +56,29 @@ def get_observations(place):
 
     return observations
 
-
 # for now this is a simple average.
 # however, the observations aren't necessarily spaced equally throughout
 # the day. do something more sophisticated later!
-def get_daily_temp(place):
-    observations = get_observations(place)
+def get_daily_temp(place, start_time, end_time):
+    observations = get_observations(place, start_time, end_time)
 
-    temps = [temp for (timestamp, temp) in observations]
+    temps = [temp for (timestamp, temp) in observations if temp]
     average = list_average(temps)
     average_fahrenheit = average * 1.8 + 32
 
     return average_fahrenheit
 
-# This currently calculates the average temperature for the current day
-def get_historical_temps(place):
-    historical_data_url = "{data_url}/csv/{id}/{month}{day}.csv".format(
-        data_url=DATA_URL,
-        id=place['USAF'] + "-" + place['WBAN'],
-        month='{:02}'.format(END_TIME.month),
-        day='{:02}'.format(END_TIME.day)
-    )
-    response = requests.get(historical_data_url)
-    lines = (line.decode('utf-8') for line in response.iter_lines())
-    reader = csv.reader(lines)
-    next(reader, None)  # skip the headers
-
-    temps_by_year = {}
-    for line in reader:
-        [year, hour, temp] = list(line)
-        try:
-            temps_by_year[year] = temps_by_year[year] + [int(temp)]
-        except KeyError:
-            temps_by_year[year] = [int(temp)]
-
-    historical_average_temps = {
-        year: (list_average(temps) * 0.18 + 32)
-        for year, temps in temps_by_year.items()
-    }
-
-    return historical_average_temps
+def get_historical_temps(place, start_time, end_time):
+    place_id = place['USAF'] + '-' + place['WBAN']
+    # first get full days of temps
+    df = _get_date_range_temps(place_id, 
+            start.date(), end.date())
+    # filter to necessary hours
+    return df[df.current_date.between(start, end)]
 
-
-def write_tweet(place, daily_temp, historical_temps):
-    total_years = len(historical_temps)
-    warmer_years = [temp for year, temp in historical_temps.items() if temp < daily_temp]
-    percent_warmer = len(warmer_years) / len(historical_temps) * 100
+def write_tweet(place, end_time, daily_temp, historical_temps):
+    year_warmer = historical_temps.groupby('year').temp.mean() > daily_temp
```

Comment:
Oof just realized this won't quite work on Jan 1! Minor issue. will need to think about fix later

Reply:
A related issue will be on March 1st since the last 24 hours in some years will include feb 29 and other years feb 28. @beamalsky how do you feel about letting this issue get resolved and then we open a new issue to deal with these subtleties?